tests/neg/i13019.scala

```diff
@@ -10,4 +10,4 @@ class Ok2C extends Ok2 { override var i: Int = 1 }
 
 // was: variable i of type Int cannot override a mutable variable
 trait NotOk {var i: Int}
-class NotOkC extends NotOk { override var i: Int = 1 }
+class NotOkC extends NotOk { override var i: Int = 1 } // error
```

Comment:
If this breaks it means we're not really aligned with Scala 2 and we need to re-open https://github.com/lampepfl/dotty/issues/13019 /cc @som-snytt . The Scala 2 logic here is https://github.com/scala/scala/commit/9b59f5f9530d54c917479c6bf44aa3007ba0a2df

Reply:
In my tweak I didn't try trait params. I'll take another look at a fix. I haven't taken the trait params coursera yet.